src/cpp/fastdds/xtypes/dynamic_types/DynamicTypeImpl.hpp

```diff
@@ -96,6 +96,11 @@ class DynamicTypeImpl : public virtual traits<DynamicType>::base_type
         return members_;
     }
 
+    TypeDescriptorImpl& get_descriptor() noexcept
+    {
+        return type_descriptor_;
+    }
+
```

Comment:
`DynamicType` is an invariable class and also its internal `TypeDescriptor`. This function have to be removed.